src/briefcase/platforms/linux/appimage.py

```diff
@@ -199,6 +199,11 @@ def build_app(self, app: AppConfig, **kwargs):
                 # so this effectively silences a bunch of warnings that can't
                 # be easily resolved by the end user.
                 env["DISABLE_COPYRIGHT_FILES_DEPLOYMENT"] = "1"
+                # AppImages do not run natively within a Docker container. This
+                # treats the AppImage like a self-extracting executable. Using
+                # this environment variable instead of --appimage-extract-and-run
+                # is necessary to ensure AppImage plugins are extracted as well.
+                env["APPIMAGE_EXTRACT_AND_RUN"] = "1"
```

Comment:
I needed another environment variable here: `env['ARCH'] = 'x86_64'`. Without it, it got complaints about multiple architectures in the AppImage when building a QT test app. This is when running on an x86 machine, so it's not M1 leakage; but it was building on Docker under macOS, so maybe something is leaking there. Either way, it can't hurt to add an explicit `env['ARCH'] = self.host_arch`.

Reply:
I added `env['ARCH'] = self.host_arch`.....but I think I'm a bit dubious of this approach.

1) Do you know where this requirement is coming from?
Neither `linuxdeploy` nor the Qt plugin sources directly reference this environment variable. That implies to me some dependency is pulling it in and leverages `ARCH`.....but without knowing what, I don't know what it's using it for.

2) If the problem is happening inside Docker, do we really want to use the arch of the host?
I'm not too sure of all the permutations possible but if the host could possibly report something other than `x86_64` while that's what the docker container is using, should we be sourcing the arch from the container instead?

3) Given these nebulous circumstances, I'm particularly inclined to let users override whatever Briefcase sets this to.
So, `env.setdefault("ARCH", arch)` instead.

Reply:
It looks like it's the `appimagetool` plugin, which AFAIK is bundled internally with `linuxdeploy`. Here's the log excerpt from an AppImage being built *without* the qt plugin enabled, on macOS in Docker, on an x86_64 machine:
```
-- Deploying desktop files -- 
Deploying desktop file /app/appimage/PySide6 Test/PySide6 Test.AppDir/com.example.pyside6test.desktop 

-- Copying files into AppDir -- 
Copying file /app/appimage/PySide6 Test/PySide6 Test.AppDir/com.example.pyside6test.desktop to /app/appimage/PySide6 Test/PySide6 Test.AppDir/usr/share/applications/com.example.pyside6test.desktop

-- Deploying files into AppDir root directory -- 
Deploying files to AppDir root using desktop file: /app/appimage/PySide6 Test/PySide6 Test.AppDir/usr/share/applications/com.example.pyside6test.desktop
Deploying desktop file to AppDir root: /app/appimage/PySide6 Test/PySide6 Test.AppDir/usr/share/applications/com.example.pyside6test.desktop 
Creating symlink for file /app/appimage/PySide6 Test/PySide6 Test.AppDir/usr/share/applications/com.example.pyside6test.desktop in/as /app/appimage/PySide6 Test/PySide6 Test.AppDir
Deploying icon to AppDir root: /app/appimage/PySide6 Test/PySide6 Test.AppDir/usr/share/icons/hicolor/128x128/apps/pyside6test.png 
Creating symlink for file /app/appimage/PySide6 Test/PySide6 Test.AppDir/usr/share/icons/hicolor/128x128/apps/pyside6test.png in/as /app/appimage/PySide6 Test/PySide6 Test.AppDir
Deploying AppRun symlink for executable in AppDir root: /app/appimage/PySide6 Test/PySide6 Test.AppDir/usr/bin/com.example.pyside6test 
Creating symlink for file /app/appimage/PySide6 Test/PySide6 Test.AppDir/usr/bin/com.example.pyside6test in/as /app/appimage/PySide6 Test/PySide6 Test.AppDir/AppRun

-- Running output plugin: appimage -- 
[appimage/stdout] Found appimagetool: /tmp/appimage_extracted_178f157ec37b2710f37a5a6e5bb1323f/plugins/linuxdeploy-plugin-appimage/usr/bin/appimagetool
[appimage/stdout] 
[appimage/stderr] Running command: /tmp/appimage_extracted_178f157ec37b2710f37a5a6e5bb1323f/plugins/linuxdeploy-plugin-appimage/usr/bin/appimagetool "/app/appimage/PySide6 Test/PySide6 Test.AppDir" "-g"
[appimage/stderr] 
[appimage/stdout] WARNING: appstreamcli command is missing, please install it if you want to use AppStream metadata
[appimage/stderr] appimagetool, continuous build (commit 7f9be69), build <local dev build> built on 2022-07-16 02:23:34 UTC
[appimage/stderr] More than one architectures were found of the AppDir source directory "/app/appimage/PySide6 Test/PySide6 Test.AppDir"
[appimage/stderr] A valid architecture with the ARCH environmental variable should be provided
[appimage/stderr] e.g. ARCH=x86_64 appimagetool ...
ERROR: Failed to run plugin: appimage (exit code: 1) 
```

The process of running x86_64 Docker images on M1 hardware is still sufficiently buggy that I'm willing to put that into the "we'll make it work when they do" bucket. At least for now, I think it's reasonable to assume/require that the architecture on which you're building is the architecture you're targeting, and not provide any special handling to override; we can revisit that decision if it turns out to be more complicated that this in practice.